contrib/clojure-package/src/org/apache/clojure_mxnet/io.clj

```diff
@@ -213,14 +245,23 @@
                last-batch-handle "pad"
                data-name "data"
                label-name "label"}}]
-   (new NDArrayIter
-        (util/vec->indexed-seq data)
-        (if label (util/vec->indexed-seq label) (util/empty-indexed-seq))
-        (int data-batch-size)
-        shuffle
-        last-batch-handle
-        data-name
-        label-name))
+   (if (map? data)
```

Comment:
does it mean if data is map? what if it is not? I don't know clojure well, just want to make sure it is intended.

Reply:
Yes - it checks to see if the data is a map, if so, it is in the form of having a DataDesc associated with it and will be dispatched to the correct Java function signature and with scala interop. If it is not a map, it will dispatch to the original Java function signature without DataDesc. 

The argument checking for the correct data structures can be improved by using core.spec in Clojure. It adds gradual type checking. It is in use in the module api, but it hasn't been added in yet in this namespace. I added a line item in the TODO page for the Clojure package to capture it for later improvement work.